src/components/m-table-header.js

```diff
@@ -41,7 +41,12 @@ export class MTableHeader extends React.Component {
             <TableSortLabel
               IconComponent={this.props.icons.SortArrow}
               active={this.props.orderBy === columnDef.tableData.id}
-              direction={this.props.orderDirection || 'asc'}
+              direction={
+                this.props.iconMatchOrientation
+                  ? this.props.orderDirection || 'asc'
+                  : this.props.orderBy === columnDef.tableData.id
+                  ? this.props.orderDirection || 'asc' : 'asc'
+              }
```

Comment:
this snippet of code seems confusing to me and risks introducing unwanted behavior
could you refactor it, please?